### ISSUE TITLE
pkgsCross.ghcjs.haskell: entropy package throws error

### ISSUE THREAD

**Issue by doyougnu** [NixOS/nixpkgs#337996]
This [commit](https://github.com/NixOS/nixpkgs/pull/335932/commits/0aa968f1b699bcd5e92651f90743b79238987ee3) adds the `entropy` package to the ghcjs package set in this [PR](https://github.com/NixOS/nixpkgs/pull/335932). `entropy` finishes building, but the build log displays:

```
entropy> No uhc found
entropy> Running phase: buildPhase
entropy> testRDRAND-301/testRDRAND.c:6:24: error:
entropy>      error: invalid output constraint '=qm' in asm
entropy>         6 |      : "=r" (therand), "=qm" (err));
entropy>           |                        ^
entropy>   |
entropy> 6 |      : "=r" (therand), "=qm" (err));
entropy>   |                        ^
entropy> 1 error generated.
entropy> `emcc' failed in phase `C Compiler'. (Exit code: 1)
entropy> Result of RDRAND Test: False
entropy> Result of libc getrandom() Test: True
```

What's going on? Well `emscripten` has failed to  compile [this](https://github.com/haskell/entropy/blob/5d0c71ef2b95b2f6314fbb6a24be7bb39dba99bb/Setup.hs#L40) function in `entropy`'s `Setup.hs` file. Why has emscripten failed here? Because emscripten [does not support inline assembly](https://github.com/emscripten-core/emscripten/issues/3118) only portable C/C++. 

What to do? I believe this is an upstream issue. `entropy` needs to be patched to use a JS random number generator since `rdrand` is not portable. That is the correct fix, alternatively, the `checkRDRAND` function could be changed to trivially pass on unsupported targets such as JS and WASM.

**Comment by doyougnu** [comment]
This is a non-issue actually. 

We see the error in the build logs because `stderr` is not captured [here](https://github.com/haskell/entropy/blob/5d0c71ef2b95b2f6314fbb6a24be7bb39dba99bb/Setup.hs#L23). _That_ there is an error is actually expected. `checkRDRAND` is working properly, RDRAND is not supported on JS so it _should_ induce this error. Then this result is passed to the next check `checkGetRandom` which _is_ supported on JS and will properly set `LocalBuildInfo` for Cabal.

So the actually issue is simply that the error is not captured by the code and we are observing it in the logs even though its benign.

**Comment by sternenseemann** [comment]
cc @alexfmpe, can this be closed?

**Comment by alexfmpe** [comment]
Yeah given it's benign I'd say there being an upstream issue is enough
